src/keys.rs

```diff
@@ -0,0 +1,123 @@
+//! Key structures for Orchard.
+//!
+//! TODO: Should we have the concept of a Network here? Or make these standalone without
```

Comment:
After writing this comment, I ran into a situation where I needed this, so I added a `Chain` trait to represent this.